eclipse-che/che-dashboard
# feat: improve the workspace update flow

### What does this PR do?
Improve the workspace update flow to change **Storage Type**.

### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/21164

### Is it tested? How?

I have tested it with **quay.io/eclipse/che-dashboard:pr-486**

Steps for testing:

1. Deploy Eclipse-Che with devworkspaces.

2. Open **Create Workspace** and chose **Temporary Storage On**.
<img width="1784" alt="2022-03-24_07_12_20" src="https://user-images.githubusercontent.com/6310786/159846835-27b9cf7b-ab8b-4840-a8c9-192f18ac8388.png">

3. Create a new workspace from one of the samples.

4. Open **Workspace Details/Overview** and try to edit **Storage Type**.
<img width="1784" alt="2022-03-24_07_15_09" src="https://user-images.githubusercontent.com/6310786/159847136-c72ae705-e9c0-4688-b6cd-a37bc1b5ec9a.png">

5. Save changes and refresh the page after. **Storage Type** should be the same after refresh.

6. Open **Workspace Details/Editor** and try to add/remove next:
```
attributes:
  controller.devfile.io/storage-type: ephemeral
```
<img width="1784" alt="2022-03-24_07_15_27" src="https://user-images.githubusercontent.com/6310786/159847931-97f0cd2d-2e0c-4ee5-8b9e-cf040b749162.png">

7. Save changes and refresh the page after. **Storage Type** should be the same after refresh.

8. Create an ephemeral devworkspace from this URL
```{Eclipse-CHE}/dashboard/#/load-factory?url=https://github.com/che-samples/golang-echo-example/tree/devfile2&storageType=ephemeral```

9. Open **Workspace Details/Editor** and try to find next:
```
attributes:
  controller.devfile.io/storage-type: ephemeral
```
<img width="1784" alt="2022-03-24_07_36_45" src="https://user-images.githubusercontent.com/6310786/159849749-1bd1a5f6-a5ca-41f1-8b97-8dda34040f75.png">

#### Release Notes



#### Docs PR
